src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -5569,16 +5630,64 @@ void markMembersAsNotNull(int receiverSlot, TypeSymbol type, string memberName,
             {
                 var (parameter, _, parameterAnnotations, _) = GetCorrespondingParameter(i, parametersOpt, argsToParamsOpt, expanded);
 
+                BoundExpression argument = arguments[i];
+                InterpolatedStringHandlerData? possibleInterpolationData = argument switch
+                {
+                    BoundInterpolatedString { InterpolationData: { } data } => data,
+                    BoundBinaryOperator { InterpolatedStringHandlerData: { } data } => data,
+                    _ => null
+                };
+
+                if (possibleInterpolationData is ({ ArgumentPlaceholders.Length: > 1 }
+                                                  or { ArgumentPlaceholders.Length: 1, HasTrailingHandlerValidityParameter: false })
+                                                 and var interpolationData)
```

Comment:
If I am correctly interpreting the intent here, I think the following condition would be easier to follow:
```possibleInterpolationData is not null && possibleInterpolationData.ArgumentPlaceholders.Length > (possibleInterpolationData.HasTrailingHandlerValidityParameter ? 1 : 0)```. Then the block can keep using ```possibleInterpolationData```.